banyan/tests/build_check.rs

```diff
@@ -26,26 +26,29 @@ fn build_stream(t: TestTree) -> anyhow::Result<bool> {
 }
 
 /// checks that stream_filtered returns the same elements as filtering each element manually
-fn compare_filtered(t: TestTree, range: Range<u64>) -> anyhow::Result<bool> {
+fn compare_filtered(t: TestTree, filter: TestFilter) -> anyhow::Result<bool> {
     let (tree, txn, xs) = t.tree()?;
     let actual = txn
-        .iter_filtered(&tree, OffsetRangeQuery::from(range.clone()))
+        .iter_filtered(&tree, filter.query())
         .collect::<anyhow::Result<Vec<_>>>()?;
     let expected = xs
         .iter()
         .cloned()
         .enumerate()
         .map(|(i, (k, v))| (i as u64, k, v))
-        .filter(|(offset, _, _)| range.contains(offset))
+        .filter(|triple| filter.contains(triple))
```

Comment:
you can't pass function als values in rust, right?

```rs
.filter(filter::contains)
```
would be nice.

Reply:
I thought so as well, but rustc does not like it. I guess Filter::contains would work but filter.contains does not.